src/API/Middlewares/HttpExceptionsMiddleware.ts

```diff
@@ -1,23 +1,33 @@
-import * as Koa from 'koa'
-
 import { IllegalArgumentException, NotFoundException } from '@po.et/poet-js'
+import { cond, T } from 'ramda'
 
-export const HttpExceptionsMiddleware: Koa.Middleware = async (context: any, next: () => Promise<any>) => {
+export const HttpExceptionsMiddleware = async (context: any, next: () => Promise<any>) => {
   try {
     await next()
   } catch (exception) {
-    if (exception instanceof IllegalArgumentException) {
-      context.body = exception.message
-      context.status = 422
-    } else if (exception instanceof NotFoundException) {
-      context.body = exception.message
-      context.status = 404
-    } else {
-      if (context.logger)
-        context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
+    const { body, status } = getCodeError(exception)
+    context.body = body
+    context.status = status
 
-      context.body = 'Internal Server Error'
-      context.status = 503
-    }
+    executeLogger(context, exception, status)
   }
 }
+
+export const executeLogger = (context: any, exception: Error, status: number) => {
+  if (context.logger && (status !== 422 && status !== 404))
```

Comment:
If we extract the `(status !== 422 && status !== 404)` to a function, we could unit test it, and it'd also give us readability by declaring intention in the function's name.

For example:

```ts
const isUnexpectedErrorStatusCode = (statusCode: number) => status !== 422 && status !== 404

export const executeLogger = (context: any, exception: Error) => {
  if (context.logger && isUnexpectedErrorStatusCode(context.status))
    context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
}
```

Reply:
done